src/sentry/quotas/redis.py

```diff
@@ -126,33 +133,59 @@ def get_value_for_result(result):
             results,
         )
 
+    def get_refunded_quota_key(self, key):
+        return 'r:{}'.format(key)
+
+    def refund(self, project, key=None, timestamp=None):
+        if timestamp is None:
+            timestamp = time()
+
+        quotas = self.get_quotas_with_limits(project, key=key)
+
+        if not quotas:
+            return
+
+        client = self.cluster.get_local_client_for_key(six.text_type(project.organization_id))
+        pipe = client.pipeline()
+
+        for quota in quotas:
+            shift = project.organization_id % quota.window
+            # kind of arbitrary, but seems like we don't want this to expire til we're
+            # sure the window is over?
+            expiry = self.get_next_period_start(quota.window, shift, timestamp) + self.grace + 5
```

Comment:
A little confused by this — what's the benefit of extending the grace period by 5?

Reply:
i just wanted to make sure the refunds didn't expire until after the quotas, but this is maybe unnecessary or could just be 1 .. ? 

side note, i think this is actually incorrect here since expiry is a unix timestamp and i need it to either be time in seconds or a timedelta http://redis-py.readthedocs.io/en/latest/#redis.StrictRedis.expire, so i'll fix that

Reply:
Ah yeah, it should probably be `EXPIREAT` like the Lua script, good catch.

I think it's OK just to use the grace period here, since as long as as `EXPIREAT` is used, the quota key and refund key should be evicted at the same time. The grace period is only there to ensure there's a bit of leeway in case the clock of the client (process handling the web request) is out of sync with the Redis server.